crates/matrix-sdk/src/room/mod.rs

```diff
@@ -410,28 +410,22 @@ impl Room {
         }
     }
 
-    async fn request_encryption_state(&self) -> Result<Option<RoomEncryptionEventContent>> {
-        if let Some(mutex) = self
-            .client
-            .inner
-            .encryption_state_request_locks
-            .get(self.inner.room_id())
-            .map(|m| m.clone())
-        {
+    async fn request_encryption_state(&self) -> Result<()> {
+        let mut map = self.client.inner.encryption_state_request_locks.lock().await;
```

Comment:
This now duplicates the logic of `request_members()` \o/ Can we factor it out, one way or another?

Reply:
Yeah we likely should have some `DedupMutex` container thingy which does this. But If you don't mind I'll leave this for another PR to not delay this one further.